### PR TITLE
Missed Christmas

### PR DESCRIPTION
**Pull request by The-Anjelo-Guy**
Mall Santa profession and Santa items

#### Summary
Content  "Added Mall Santa as a profession and some other Santa items"

#### Purpose of change
I thought it would be cool to have a Mall Santa profession in Christmas, even though I'm probably too late.

### CODE REVIEW

**Comment by Termineitor244** [comment]
Most of the items you are using for the profession don't actually exist... And I would know it since I added the ones that are in the game right now! Also, if you are planning to add them in this PR, then most of them should be just variants of existing items, like the necklace or the wristwatch, like how I added vampire rings/necklace and the like, they are small and similar enough to be represented with a variant.

**Comment by The-Anjelo-Guy** [comment]
> Most of the items you are using for the profession don't actually exist... And I would know it since I added the ones that are in the game right now! Also, if you are planning to add them in this PR, then most of them should be just variants of existing items, like the necklace or the wristwatch, like how I added vampire rings/necklace and the like, they are small and similar enough to be represented with a variant.
> 
> Oh, and if you add the new items/variants please add them to the cosplays itemgroups like the rest of the santa items, there are some new santa itemdrops too for Christmas so those groups are a good idea for these items too.

Hey hello gentleman. I'll be adding the items soon once I get some free time. Don't worry, I'll do everything properly.

**Comment by GuardianDll** [comment]
Please don't add santa_knifr

**Comment by The-Anjelo-Guy** [comment]
> Please don't add santa_knife

It is a novelty item that Mrs. Claus use in the kitchen, but I'm open for ideas of weaponry for them other than a large novelty candy cane.

File: data/json/items/armor/boots.json [hunk 1]
```diff
@@ -2157,6 +2203,48 @@
       }
     ]
   },
+  {
+    "id": "santa_shoes",
+    "type": "ARMOR",
+    "name": { "str": "pair of Santa shoes", "str_pl": "pairs of Santa shoes" },
+    "description": "A delightful pair of comfortable red and white shoes with Christmas adornments.  For either Mrs. Claus or Santa style and comfort.",
```
Comment (Qrox) [inline data/json/items/armor/boots.json:2210]: FYI you can suppress this and similar false positives with the following changes:

```suggestion
    "description": {
        "str": "A delightful pair of comfortable red and white shoes with Christmas adornments.  For either Mrs. Claus or Santa style and comfort.",
        "//NOLINT(cata-text-style)": "Mrs. does not end with a period"
    },
```

File: data/json/items/armor/boots.json [hunk 2]
```diff
@@ -102,6 +102,52 @@
     "proportional": { "weight": 0.75, "volume": 0.75, "price": 0.5 },
     "extend": { "flags": [ "UNDERSIZE" ] }
   },
+  {
+    "id": "santa_boots",
+    "type": "ARMOR",
+    "name": { "str": "pair of Santa boots", "str_pl": "pairs of Santa boots" },
+    "description": "A tough pair of black leather boots, trimmered with white faux fur.  For tough everyday work at the North Pole.",
+    "copy_from": "boots",
```
Comment (Qrox) [inline data/json/items/armor/boots.json:110]:
```suggestion
    "copy-from": "boots",
```

File: data/json/items/armor/boots.json [hunk 3]
```diff
@@ -2157,6 +2203,51 @@
...
   },
+  {
+    "id": "santa_shoes",
+    "type": "ARMOR",
+    "name": { "str": "pair of Santa shoes", "str_pl": "pairs of Santa shoes" },
+    "description": {
+      "str": "A delightful pair of comfortable red and white shoes with Christmas adornments.  For either Mrs. Claus or Santa style and comfort.",
+      "//NOLINT(cata-text-style)": "Mrs. does not end with a period"
+    },
+    "copy_from": "sneakers",
```
Comment (Qrox) [inline data/json/items/armor/boots.json:2214]:
```suggestion
    "copy-from": "sneakers",
```

File: data/json/items/armor/boots.json [hunk 4]
```diff
@@ -2641,6 +2732,25 @@
     "proportional": { "weight": 0.75, "volume": 0.75, "price": 0.5 },
     "extend": { "flags": [ "UNDERSIZE" ] }
   },
+  {
+    "id": "santa_socks",
+    "type": "ARMOR",
+    "name": { "str": "pair of Santa socks", "str_pl": "pairs of Santa socks" },
+    "description": "A pair of comfortable red socks lightly trimmered with white faux fur.  Kind like the ones you'd hang on your fireplace, but wearable and lightweight.",
+    "copy_from": "socks",
```
Comment (Qrox) [inline data/json/items/armor/boots.json:2740]:
```suggestion
    "copy-from": "socks",
```

File: data/json/items/armor/boots.json [hunk 5]
```diff
@@ -3145,6 +3255,14 @@
         "weight": 25,
         "color": "green",
         "append": true
+      },
+      {
+        "id": "santa_stockings",
+        "name": { "str": "pair of Christmas stockings", "str_pl": "pairs of Christmas stockings" },
+        "description": "A comfortable pair of red, white and green stockings with Christmas patterns on it, for Mrs. Claus style.",
```
Comment (Qrox) [inline data/json/items/armor/boots.json:3262]:
```suggestion
        "description": {
          "str": "A comfortable pair of red, white and green stockings with Christmas patterns on it, for Mrs. Claus style.",
          "//NOLINT(cata-text-style)": "Mrs. does not end with a period"
        },
```

File: data/json/items/armor/jewelry.json [hunk 6]
```diff
@@ -39,6 +39,21 @@
     "color": "brown",
     "flags": [ "NO_WEAR_EFFECT" ]
   },
+  {
+    "id": "santa_ear",
+    "type": "ARMOR",
+    "name": { "str": "pair of candycane earrings", "str_pl": "pairs of candycane earrings" },
+    "description": "A festive, lightweight and simple pair of earrings made of string and tiny plastic candycanes.  For Mrs. Claus style.",
```
Comment (Qrox) [inline data/json/items/armor/jewelry.json:46]:
```suggestion
    "description": {
      "str": "A festive, lightweight and simple pair of earrings made of string and tiny plastic candycanes.  For Mrs. Claus style.",
      "//NOLINT(cata-text-style)": "Mrs. does not end with a period"
    },
```

File: data/json/professions.json [hunk 7]
```diff
@@ -2756,6 +2756,60 @@
       "female": { "entries": [ { "item": "bra" }, { "item": "panties" }, { "item": "gold_ear" } ] }
     }
   },
+  {
+    "type": "profession",
+    "id": "santa",
+    "name": { "male": "Mall Santa", "female": "Mall Mrs. Claus" },
+    "description": "Everything was set for Christmas, you where just waiting for the call.  The children's happiness and joy where lost due to the Cataclysm, but you are decided to keep the Christmas Spirit alive, no matter what.  At least, all that time making toys and carrying them will be of use.",
```
Comment (harakka) [inline data/json/professions.json:2763]: This does not depict someone working as a mall santa. Neither does the skill set.

**Review by GuardianDll — CHANGES_REQUESTED** [review]
copy from our discord discussion:

santa_undershirt, santa_socks, santa_briefs, santa_bra, santa_panties should be gone, unless you want to make them a mundane items, not related to santa set

everything else should go into variants of mundane items with weight 0 (so they don't spawn naturally, it can be reverted if we decide to make more variants for said clothes), and itemgroups should call this variant

**Comment by The-Anjelo-Guy** [comment]
I think I done it properly now. The Santa undergarment and Santa knife are gone, fixed the description and skillset for the Mall Santa profession, fixed volume, weight and material of the other items, instead of doing variants, for logistics and also because these items have different materials and physical values than their variant-intended base items. I would like to thank Guardiandl and Harakka for their help and support in this PR, nothing would be achieved without them.

**Review by GuardianDll — APPROVED** [review]
looks good for me now, test error seems unrelated

**Pull request merged**